packages/edit-post/src/store/selectors.js

```diff
@@ -105,11 +105,20 @@ export function getPreferences( state ) {
  *
  * @return {*} Preference Value.
  */
-export function getPreference( state, preferenceKey, defaultValue ) {
-	const preferences = getPreferences( state );
-	const value = preferences[ preferenceKey ];
-	return value === undefined ? defaultValue : value;
-}
+export const getPreference = createRegistrySelector(
+	( select ) => ( state, preferenceKey, defaultValue ) => {
+		if ( preferenceKey === 'hiddenBlockTypes' ) {
+			return select( preferencesStore ).get(
+				'core/edit-post',
+				preferenceKey
+			);
+		}
+
+		const preferences = getPreferences( state );
```

Comment:
I need to do a bit more work on back compat here and make sure `getPreferences` has any migrated preferences merged into it.

Reply:
Would we then deprecate `getPreference`?

Reply:
Once everything is migrated I think it can be deprecated, yep 😄